src/main/java/space/npstr/magma/AudioStackLifecyclePipeline.java

```diff
@@ -113,10 +114,31 @@ protected void hookOnNext(final LifecycleEvent event) {
         }
     }
 
+    @CheckReturnValue
+    public Collection<WebsocketConnectionState> getAudioConnectionStates() {
+        return this.audioStacks.entrySet().stream()
+                .flatMap(outerEntry -> {
+                    final String userId = outerEntry.getKey();
+                    return outerEntry.getValue().entrySet().stream()
+                            .map(innerEntry -> {
+                                final String guildId = innerEntry.getKey();
+                                final AudioStack audioStack = innerEntry.getValue();
+                                return MagmaWebsocketConnectionState.builder()
+                                        .member(MagmaMember.builder()
+                                                .userId(userId)
+                                                .guildId(guildId)
+                                                .build())
+                                        .phase(audioStack.getConnectionPhase())
+                                        .build();
+                            });
+                })
+                .collect(Collectors.toList());
+    }
+
     @CheckReturnValue
     private AudioStack getAudioStack(final LifecycleEvent lifecycleEvent) {
         return this.audioStacks
-                .computeIfAbsent(lifecycleEvent.getUserId(), __ -> new HashMap<>())
+                .computeIfAbsent(lifecycleEvent.getUserId(), __ -> new ConcurrentHashMap<>())
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. ([squid:S00117](https://sonarcloud.io/organizations/napstr-github/rules#rule_key=squid%3AS00117))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=space.npstr.magma%3Amagma&pullRequest=15&issues=AWatMHETq3jbRxE0h66O&open=AWatMHETq3jbRxE0h66O)

[ISSUE_KEY]: AWatMHETq3jbRxE0h66O